src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1297,24 +1264,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
         if (node.tag === HostRoot) {
           const root: FiberRoot = (node.stateNode: any);
           scheduleRoot(root, priorityLevel);
-          // Depending on the priority level, either perform work now or
-          // schedule a callback to perform work later.
-          switch (priorityLevel) {
-            case SynchronousPriority:
-              performWork(SynchronousPriority, null);
-              return;
-            case TaskPriority:
-              // TODO: If we're not already performing work, schedule a
-              // deferred callback.
-              return;
```

Comment:
This would schedule an unnecessary deferred callback if you call setState in componentDidMount inside a sync render.

Would be nice to have a test for that like Dan's ReactIncrementalPerf-test.js.snap that spawned these other changes.